src/directed/fringe.rs

```diff
@@ -138,8 +138,8 @@ where
                         }
                     }
                 }
-                if !remove(&mut later, &n) {
-                    remove(&mut now, &n);
+                if later.remove(n).is_some() {
+                    now.remove(n);
```

Comment:
How is that equivalent? The former removes the first element equal to `n` from `later` and `now` (regardless of its index), while the latter seems to remove the element at index `n` from `later` and `now`.

Reply:
oh im sorry corrected